shopfloor/services/zone_picking.py

```diff
@@ -535,23 +552,34 @@ def _scan_source_location(
             )
         else:
             # if no move line, narrow the list of move lines on the scanned location
-            response = self._list_move_lines(location)
-            message = self.msg_store.location_empty(location)
+            response = self._list_move_lines(location, package=package)
+            # Location is empty: invite the user to scan a package if the option is enabled
+            if scan_location_or_pack_first:
+                message = self.msg_store.location_empty_scan_package(location)
+            else:
+                message = self.msg_store.location_empty(location)
```

Comment:
The wrong location has been scanned. We should list again all possibilities of the zone and raise an error

```suggestion
            response = self._list_move_lines(self.zone_location)
            message = self.msg_store.wrong_record(location)
```

Reply:
Fixed